src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -415,14 +435,31 @@ protected function getAliasedPivotColumns()
         // relationships when they are retrieved and hydrated into the models.
         $columns = [];
 
-        foreach (array_merge($defaults, $this->pivotColumns) as $column) {
+        foreach (array_merge($defaults, $this->getPivotModelColumns(), $this->pivotColumns) as $column) {
             $columns[] = $this->table.'.'.$column.' as pivot_'.$column;
         }
 
         return array_unique($columns);
     }
 
     /**
+     * Retrieve necessary columns from the custom pivot model.
+     *
+     * @return array
+     */
+    protected function getPivotModelColumns()
+    {
+        $columns = [];
+
+        if ($this->using) {
+            $customPivotClass = $this->using;
+            $columns = $customPivotClass::getPivotColumns();
```

Comment:
Why you have this new way of specifying which columns to load? Why not just use the traditional belongsToMany->with()?